packages/datatrak-web/src/utils/detectDevice.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+// detect if the device is a mobile device (without relying on screen size)
+export const getIsMobileDevice = () => {
+  const regex = /Mobi|Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i;
```

Comment:
Out of interest, where did you get the regex list? When I googled it the lists I saw were often much longer? eg. https://stackoverflow.com/questions/11381673/detecting-a-mobile-browser. Anyway overall I don't think it matters too much as if we don't prompt one user once in a while.

Reply:
Also from a stackoverflow. I think it should theoretically pickup most mobile devices based on this one though

Reply:
If from SO, remember to attribute! SO answers are shared under the [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0) licence, and it’s super easy to forget that really we should be crediting + linking, and indicating if we made changes